components/brave_wallet/browser/ethereum_provider_impl.cc

```diff
@@ -311,22 +316,12 @@ void EthereumProviderImpl::OnGetNetworkAndDefaultKeyringInfo(
                          keyring_info->account_infos, from)) {
     // Set chain_id to current chain_id.
     tx_data_1559->chain_id = chain->chain_id;
-    // If the chain id is not known yet, then get it and set it first
-    if (tx_data_1559->chain_id == "0x0" || tx_data_1559->chain_id.empty()) {
-      json_rpc_service_->GetChainId(
-          mojom::CoinType::ETH,
-          base::BindOnce(
-              &EthereumProviderImpl::ContinueAddAndApprove1559Transaction,
-              weak_factory_.GetWeakPtr(), std::move(callback), std::move(id),
-              std::move(tx_data_1559), from, origin));
-    } else {
-      GetAllowedAccounts(
-          false,
-          base::BindOnce(&EthereumProviderImpl::
-                             ContinueAddAndApprove1559TransactionWithAccounts,
-                         weak_factory_.GetWeakPtr(), std::move(callback),
-                         std::move(id), std::move(tx_data_1559), from, origin));
-    }
+    GetAllowedAccounts(
+        false,
+        base::BindOnce(&EthereumProviderImpl::
+                           ContinueAddAndApprove1559TransactionWithAccounts,
+                       weak_factory_.GetWeakPtr(), std::move(callback),
+                       std::move(id), std::move(tx_data_1559), from, origin));
   } else {
     if (!tx_data_1559) {
```

Comment:
This might be always false for quite some time

Reply:
yep, it should be removed because there are already `!tx_data_1559` check before this if/else

Reply:
fixed and squashed.